File: 957-minimum-add-to-make-parentheses-valid/minimum-add-to-make-parentheses-valid.java
class Solution {
    public int minAddToMakeValid(String s) {
        Stack<String>st=new Stack<>();
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='(')
            {
                st.push("(");
            }
            else if(s.charAt(i)==')')
            {
                if(!st.empty() && st.peek()=="(")
                {
                    st.pop();

                }
                else{
                    st.push(")");
                }
            }
        }
        return st.size();
        
    }
}

File: 957-minimum-add-to-make-parentheses-valid/minimum-add-to-make-parentheses-valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        vector<string >ans;

        for(int i=0;i<s.size();i++){
            if(s[i]=='(')
            {
                ans.push_back("(");
            }
            else if(s[i]==')')
            {

                if(!ans.empty() &&  ans[ans.size()-1]=="("){
                  
                        ans.pop_back();

                    }
                
                    else{
                    ans.push_back(")");
                }
               
            }
        }
        return ans.size();
        
    }
};